cratedb_retention/store.py

```diff
@@ -198,3 +208,22 @@ def query(self, statement):
         rows = cursor.mappings().fetchall()
         records = [dict(row.items()) for row in rows]
         return records
+
+    def exists(self, policy: RetentionPolicy):
+        """
+        Check if retention policy for specific table already exists.
+        """
+        table = self.table
+        selectable = sa.select(table).where(
+            table.c.table_schema == policy.table_schema,
+            table.c.table_name == policy.table_name,
+        )
```

Comment:
It can make sense to have multiple policies for one table. Especially to combine the reallocate strategy with others.
Example: For table `x`, I want to reallocate partitions after 30 days from a hot to a cold node, and after 60 days I want to permanently delete it (or snapshot with deletion).

But indeed it doesn't make sense to have a combination of the snapshot + delete strategies on the same table, as both will erase the data.

A reallocate strategy with a shorter retention period than a delete/snapshot strategy makes sense, though.

Reply:
Hi Niklas.

I was having roughly about the same thoughts, so I am happy you are referring to a specific example. Maybe we can track the new insights at GH-26, in order to come up with a better plan how to handle the constraints more specifically?

WIth kind regards,
Andreas.

Reply:
Sounds good, let's save it for another iteration 👍